Documentation/contributors/design-proposals/operator-validation-library.md

```diff
@@ -53,3 +53,191 @@ There is also a Command Line Tool that serves as an example of how a user can in
+```
+
+
+### Commands 
+**For individual files**: 
+
+```
+$ operator-verify csv /path/to/csv.yaml
+```
+
+Validates the given CSV yaml file against OLM type and reports errors and warnings as described in the `List of Errors` section below.
+
+Similarly, 
+
+```
+$ operator-verify crd /path/to/crd.yaml
+```
+
+and 
+
+```
+$ operator-verify package /path/to/package.yaml
+```
+
+validates CRD and package yaml, respectively. The `crd` command can accept and validate both a single yaml file and a bunch of CRDs at once and report a separate log for each.
+
+**For manifest bundle**:
+
+```
+$ operator-verify manifest /path/to/bundle
+```
+
+Here `path/to/bundle` is a directory structure as per [manifest format](https://github.com/operator-framework/operator-registry#manifest-format). It can be either be flattened or nested with different versions of yaml files. This command reports errors and/or warnings for each file in the bundle.
+
+## Examples
+
+```
+$ operator-verify csv csv.yaml
+```
+
+Output of the `csv` command against a valid sample csv yaml with some missing optional fields/struct:
+
+```
+Warning: Optional Field Missing (ObjectMeta.GenerateName)
+Warning: Optional Field Missing (ObjectMeta.SelfLink)
+Warning: Optional Field Missing (ObjectMeta.UID)
+Warning: Optional Field Missing (ObjectMeta.ResourceVersion)
+Warning: Optional Field Missing (ObjectMeta.Generation)
+Warning: Optional Struct Missing (ObjectMeta.CreationTimestamp)
+Warning: Optional Field Missing (ObjectMeta.DeletionTimestamp)
+Warning: Optional Field Missing (ObjectMeta.DeletionGracePeriodSeconds)
+Warning: Optional Field Missing (ObjectMeta.Labels)
+Warning: Optional Field Missing (ObjectMeta.OwnerReferences)
+Warning: Optional Field Missing (ObjectMeta.Initializers)
+Warning: Optional Field Missing (ObjectMeta.Finalizers)
+Warning: Optional Field Missing (ObjectMeta.ClusterName)
+Warning: Optional Field Missing (Spec.CustomResourceDefinitions.Required)
+Warning: Optional Struct Missing (Spec.APIServiceDefinitions)
+Warning: Optional Field Missing (Spec.NativeAPIs)
+Warning: Optional Field Missing (Spec.MinKubeVersion)
+Warning: Optional Field Missing (Spec.Provider.URL)
+Warning: Optional Field Missing (Spec.Replaces)
+Warning: Optional Field Missing (Spec.Annotations)
+csv.yaml is verified.
+```
+
+Omitting `Spec.InstallStrategy.StrategyName`, one of the mandatory fields, yields
+
+```
+Error: Mandatory Field Missing (Spec.InstallStrategy.StrategyName)
+Populate all the mandatory fields missing from csv.yaml file.
+```
+
+in addition to the warnings shown above.
+
+To ignore the warnings, we have `-i` flag available.
+
+```
+$ operator-verify -i csv csv.yaml
+```
+
+`crd` and `package` commands works in a similar way as the `csv`. The only difference being `crd` can also accept a directory containing a bunch of CRDs to validate. The `manifest` command returns a log similar to the one shown above for `csv` for each file in the manifest bundle.
```

Comment:
Same comment as above about validating directories. What prevents you from having a `--csv-only` flag for the manifest command? You can have multiple CSVs in your bundle